LLVM assertion error on release-0.6 while building docs

Travis is triggering an LLVM assertions on release-0.6 while building the docs: https://travis-ci.org/JuliaLang/julia/jobs/309198066

```
julia: /home/travis/build/JuliaLang/julia/deps/srccache/llvm-3.9.1/lib/IR/Instruction.cpp:30: llvm::Instruction::Instruction(llvm::Type*, unsigned int, llvm::Use*, unsigned int, llvm::Instruction*): Assertion `BB && "Instruction to insert before is not in a basic block!"' failed.
signal (6): Aborted
while loading /home/travis/build/JuliaLang/julia/doc/make.jl, in expression starting on line 5
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x2b1b157f0bf5)
__assert_fail at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_ZN4llvm11InstructionC2EPNS_4TypeEjPNS_3UseEjPS0_ at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
_ZN4llvm8LoadInstC2EPNS_4TypeEPNS_5ValueERKNS_5TwineEbjNS_14AtomicOrderingENS_20SynchronizationScopeEPNS_11InstructionE at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
_ZN4llvm8LoadInstC1EPNS_4TypeEPNS_5ValueERKNS_5TwineEbjPNS_11InstructionE at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
unknown function (ip: 0x2b1b1674541f)
unknown function (ip: 0x2b1b16748d28)
_ZN4llvm15PromoteMemToRegENS_8ArrayRefIPNS_10AllocaInstEEERNS_13DominatorTreeEPNS_15AliasSetTrackerEPNS_15AssumptionCacheE at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
unknown function (ip: 0x2b1b16733bd8)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /home/travis/build/JuliaLang/julia/usr/bin/../lib/libLLVM-3.9.so (unknown line)
operator() at /home/travis/build/JuliaLang/julia/src/jitlayers.cpp:460 [inlined]
_M_invoke at /usr/include/c++/5/functional:1857
operator() at /usr/include/c++/5/functional:2267 [inlined]
addModuleSet<llvm::SmallVector<std::unique_ptr<llvm::Module>, 1u>, llvm::RTDyldMemoryManager*, std::unique_ptr<llvm::orc::LambdaResolver<JuliaOJIT::addModule(std::unique_ptr<llvm::Module>)::<lambda(const string&)>, JuliaOJIT::addModule(std::unique_ptr<llvm::Module>)::<lambda(const string&)> >, std::default_delete<llvm::orc::LambdaResolver<JuliaOJIT::addModule(std::unique_ptr<llvm::Module>)::<lambda(const string&)>, JuliaOJIT::addModule(std::unique_ptr<llvm::Module>)::<lambda(const string&)> > > > > at /home/travis/build/JuliaLang/julia/usr/include/llvm/ExecutionEngine/Orc/IRCompileLayer.h:73 [inlined]
addModule at /home/travis/build/JuliaLang/julia/src/jitlayers.cpp:587
jl_add_to_ee at /home/travis/build/JuliaLang/julia/src/jitlayers.cpp:810 [inlined]
jl_finalize_function at /home/travis/build/JuliaLang/julia/src/jitlayers.cpp:821
getAddressForFunction at /home/travis/build/JuliaLang/julia/src/codegen.cpp:1411
jl_generate_fptr at /home/travis/build/JuliaLang/julia/src/codegen.cpp:1508
jl_compile_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:326 [inlined]
jl_call_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:354 [inlined]
jl_apply_generic at /home/travis/build/JuliaLang/julia/src/gf.c:1926
#cd#1 at ./pkg/dir.jl:32
jl_call_fptr_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/travis/build/JuliaLang/julia/src/gf.c:1926
jl_apply at /home/travis/build/JuliaLang/julia/src/julia.h:1424 [inlined]
jl_invoke at /home/travis/build/JuliaLang/julia/src/gf.c:51
update at ./pkg/pkg.jl:228
jl_call_fptr_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/travis/build/JuliaLang/julia/src/gf.c:1926
do_call at /home/travis/build/JuliaLang/julia/src/interpreter.c:75
eval at /home/travis/build/JuliaLang/julia/src/interpreter.c:242
jl_interpret_toplevel_expr at /home/travis/build/JuliaLang/julia/src/interpreter.c:34
jl_toplevel_eval_flex at /home/travis/build/JuliaLang/julia/src/toplevel.c:577
jl_parse_eval_all at /home/travis/build/JuliaLang/julia/src/ast.c:873
jl_load at /home/travis/build/JuliaLang/julia/src/toplevel.c:616
include_from_node1 at ./loading.jl:576
unknown function (ip: 0x2b1b19c916bb)
jl_call_fptr_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/travis/build/JuliaLang/julia/src/gf.c:1926
include at ./sysimg.jl:14
unknown function (ip: 0x2b1b19b2394b)
jl_call_fptr_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/travis/build/JuliaLang/julia/src/gf.c:1926
process_options at ./client.jl:305
_start at ./client.jl:371
unknown function (ip: 0x2b1b19ca04b8)
jl_call_fptr_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/travis/build/JuliaLang/julia/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/travis/build/JuliaLang/julia/src/gf.c:1926
jl_apply at /home/travis/build/JuliaLang/julia/ui/../src/julia.h:1424 [inlined]
true_main at /home/travis/build/JuliaLang/julia/ui/repl.c:127
main at /home/travis/build/JuliaLang/julia/ui/repl.c:264
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x40160e)
Allocations: 6081730 (Pool: 6080128; Big: 1602); GC: 11
make[2]: *** [html] Aborted (core dumped)
263.939448 make[2]: Leaving directory `/home/travis/build/JuliaLang/julia/doc'
make[1]: *** [docs] Error 2
make: *** [/home/travis/build/JuliaLang/julia/doc/_build/html/en/index.html] Error 2
263.939900 make[1]: Leaving directory `/home/travis/build/JuliaLang/julia
```

I haven't managed to reproduce that locally. cc: @ararslan

--- a/.travis.yml
+++ b/.travis.yml
@@ -109,6 +109,9 @@ script:
     - contrib/download_cmake.sh
     - make -C moreutils mispipe
     - make $BUILDOPTS -C base version_git.jl.phony
+    # Check that the llvm cache is not poisoned
+    - if [ -f deps/srccache/llvm-3.9.1/llvm-D37939-Mem2Reg-Also-handle-memcpy.patch-applied ]; then
+      make -C deps distclean-llvm; fi
     # capture the log, but only print it if `make deps` fails
     # try to show the end of the log first, because this log might be very long (> 4MB)
     # and thus be truncated by travis

